src/utilities/sorting/sorting.ts

```diff
@@ -0,0 +1,70 @@
+import {CARTA} from "carta-protobuf";
+
+import {ControlHeader} from "stores";
+import {ProcessedColumnData} from "utilities";
+
+export function updateSortedIndexMapFunc(
+    controlHeader: Map<string, ControlHeader>,
+    sortingInfo: {columnName: string; sortingType: CARTA.SortingType},
+    sortedIndexMap: Array<number>,
+    hasFilter: boolean,
+    numVisibleRows: number,
+    sortData: Map<number, ProcessedColumnData>
+) {
+    const dataIndex = controlHeader.get(sortingInfo.columnName)?.dataIndex;
+
+    if (dataIndex >= 0) {
+        let direction = 0;
+
+        if (sortingInfo.sortingType != null) {
+            direction = sortingInfo.sortingType ? -1 : 1;
```

Comment:
I suggest we change it to `sortingInfo.sortingType === CARTA.SortingType.Descending` for better readability.